transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultMigrationManager.java

```diff
@@ -72,7 +72,18 @@
               8,
               "Update length of invocation column on outbox for MySQL dialects only.",
               "ALTER TABLE TXNO_OUTBOX MODIFY COLUMN invocation MEDIUMTEXT",
-              Map.of(Dialect.POSTGRESQL_9, "", Dialect.H2, "")));
+              Map.of(Dialect.POSTGRESQL_9, "", Dialect.H2, "")),
+          new Migration(
+              9,
+              "Add createTime column to outbox",
+              "ALTER TABLE TXNO_OUTBOX ADD COLUMN createTime TIMESTAMP(6) NULL AFTER invocation",
```

Comment:
Timestamps aren't guaranteed 100% reliable for ordering; it's theoretically possible to write two records at exactly the same time.

I guess the question is whether that's of particular concern.  We are watching this PR with interest here, since we may have a use for it, but our use case actually doesn't care about ordering at all; we're looking to use it to throttle processing to one task per group id at a time as a way of "rationing" our batch processing capability between customers.

Reply:
I don't need to have that "super-precise" reliability in my project either, but I think it may be useful for others. 

I'd probably go with a single sequence (or auto-incremented column) for a counter but use it only in tandem with the `createTime` field in order to mitigate a potential "max-value to min-value" problem almost completely.
Need to brush up MySQL 5 capabilities for all of that though

Reply:
I've used a special sequence table for this in the past, to mimic the flexibility of oracle sequences.

Reply:
@amseager With the potential "max-value to min-value" problem, are you referring to a sequence wrapping around from its maximum value to its minimum value?

Reply:
@amseager With the potential "max-value to min-value" problem, are you referring to a sequence wrapping around from its maximum value to its minimum value?